test/sample_exact_es_percent/processor-spec.ts

```diff
@@ -0,0 +1,236 @@
+import { WorkerTestHarness } from 'teraslice-test-harness';
+import { SampleExactESPercentConfig } from '../../asset/src/sample_exact_es_percent/interfaces.js';
+import { debugLogger, pDelay, TestClientConfig } from '@terascope/job-components';
+import { makeData } from '../test_helpers.js';
+
+describe('sample_exact_es_percent', () => {
+    let harness: WorkerTestHarness;
+    const logger = debugLogger('test-logger');
+
+    // Each harness will make a get request to the mock elasticsearch client on initialization, and
+    // once every 'window_ms' the harness exists beyond that. Each request to the client will shift
+    // off and return the first element in percentArr.
+    // Each test should overwrite percentArr with a new array.
+    let percentArr: Array<number | string> = [];
+
+    const defaultClients: TestClientConfig[] = [
+        {
+            type: 'elasticsearch-next',
+            endpoint: 'default',
+            createClient: async () => ({
+                client: {
+                    get: () => {
+                        return { _source: { percent: percentArr.shift() }, found: true };
+                    }
+                },
+                logger
+            }),
+        }
+    ];
+
+    async function makeTest(config: Partial<SampleExactESPercentConfig> = { index: 'my-index', document_id: 'abc123' }, clients = defaultClients) {
+        const baseConfig = {
+            _op: 'sample_exact_es_percent',
+        };
+        const opConfig = Object.assign({}, baseConfig, config);
+        harness = WorkerTestHarness.testProcessor(opConfig, { clients });
+
+        await harness.initialize();
+
+        return harness;
+    }
+
+    afterEach(async () => {
+        if (harness) await harness.shutdown();
+    });
+
+    it('with initial percentage at 100%, should return empty array from empty array', async () => {
+        percentArr = [100];
+        harness = await makeTest();
+        const results = await harness.runSlice([]);
+
+        expect(results).toEqual([]);
+    });
+
+    it('with initial percentage at 100%, should return all the data', async () => {
+        percentArr = ['100'];
+        const data = makeData(10);
+        harness = await makeTest();
+        const results = await harness.runSlice(data);
+
+        expect(results.length).toEqual(10);
+    });
+
+    it('shuffles the data', async () => {
+        percentArr = [100];
+        const data = makeData(10);
+        harness = await makeTest();
```

Comment:
This `makeTest` kind of melts my brain.  It distracts me from the test itself identified by the `it`.  The function returns a harness right?  Can we rename this to `getHarness` or `getWorkerHarness`?